bundler/lib/bundler/cli/lock.rb

```diff
@@ -71,8 +68,11 @@ def run
         if print
           puts definition.to_lock
         else
+          file = options[:lockfile]
+          file = file ? Pathname.new(file).expand_path : Bundler.default_lockfile
```

Comment:
The file passed here needn't be a `Pathname` before https://github.com/rubygems/rubygems/pull/7047/files and probably not needed either now, but since `Bundler.default_lockfile` is a `Pathname` anyways, I think it makes sense to pass a consistent type to `Definition#write_lock` 👍.